src/build/__main__.py

```diff
@@ -94,15 +93,13 @@ def main_parser():  # type: () -> argparse.ArgumentParser
     """
     Constructs the main parser
     """
-    cwd = os.getcwd()
-    out = os.path.join(cwd, 'dist')
     parser = argparse.ArgumentParser()
     parser.add_argument(
         'srcdir',
         type=str,
         nargs='?',
         metavar='sourcedir',
-        default=cwd,
+        default='.',
```

Comment:
let's keep here the os.getcwd()

Reply:
My intention here was for the CLI to inherit the programmatic API's default arguments which IMO makes the code easier to reason about.  Can rv if you prefer to keep them separate.

Reply:
But outdir is affected here, sourcedir remains fully qualified, as is with the programmatic API.

Reply:
The srcdir is made absolute in the ProjectBuilder's `__init__` - expanding `cwd` shouldn't be necessary, though there might be (probably is :) something I'm overlooking?

What I mean by inheriting the programmatic API is, if you compare the signatures of:

```py 
class ProjectBuilder(object):
    def __init__(self, srcdir='.', config_settings=None, python_executable=sys.executable): ...
        #              ^
    def build(self, distribution, outdir=None): ...
        #                         ^
```

with the CLI arguments:

```py
parser.add_argument(
    'srcdir',
    default='.',
)
parser.add_argument(
    'outdir',
    # default implicitly None
)
```

they have the same defaults.

Reply:
In this case I actually would prefer to change both `.` to `os.getcwd()` calls. In the spirit of explicit better than implicit.

Reply:
@layday this remains the only outstanding point, any thoughts?

Reply:
hmm, I'm not sure how we'd approach ProjectBuilder defaulting to `os.getcwd()` - we'd not wanna use it as a default argument cuz then it'd be generated on import.  We could use `os.curdir` instead of `'.'` if you find that to be more explicit.

Reply:
The pythonic way of doing it is to make default None, and set os.getcwd, if None. Assign that as attribute of the class.

Reply:
That does actually tie in nicely with how `outdir` is evaluated, done.